### PR TITLE
Explain fetchTarball timeout behavior in the doc

### PR DESCRIPTION
**Pull request by mickours**
Just add some doc on the timeout mechanism of the fetchTarball builtins (because I only find this information on the IRC :))

### CODE REVIEW

File: doc/manual/expressions/builtins.xml [hunk 1]
```diff
@@ -280,8 +280,15 @@ with import (fetchTarball https://github.com/NixOS/nixpkgs-channels/archive/nixo
 
 stdenv.mkDerivation { … }
 </programlisting>
-
-    Note that when obtaining the hash with <varname>nix-prefetch-url
+    <para>The fetched tarball is chached for a certain amount of time 
+    (1 hour by default) in <literal>~/.cache/nix/tarballs/</literal>.
+    You can change the cache timeout either in command line with 
+    <literal>--option tarball-ttl <number of seconds></literal> or
```
Comment (edolstra) [inline doc/manual/expressions/builtins.xml]: `<number of seconds>` is not valid XML. You should do something like `<replaceable>ttl</replaceable>`.

File: doc/manual/expressions/builtins.xml [hunk 2]
```diff
@@ -280,8 +280,15 @@ with import (fetchTarball https://github.com/NixOS/nixpkgs-channels/archive/nixo
 
 stdenv.mkDerivation { … }
 </programlisting>
-
-    Note that when obtaining the hash with <varname>nix-prefetch-url
+    <para>The fetched tarball is chached for a certain amount of time 
+    (1 hour by default) in <literal>~/.cache/nix/tarballs/</literal>.
+    You can change the cache timeout either in command line with
```
Comment (edolstra) [inline doc/manual/expressions/builtins.xml]: `in` -> `on the`.

File: doc/manual/expressions/builtins.xml [hunk 3]
```diff
@@ -280,8 +280,15 @@ with import (fetchTarball https://github.com/NixOS/nixpkgs-channels/archive/nixo
...
 stdenv.mkDerivation { … }
 </programlisting>
-
-    Note that when obtaining the hash with <varname>nix-prefetch-url
+    <para>The fetched tarball is chached for a certain amount of time 
+    (1 hour by default) in <literal>~/.cache/nix/tarballs/</literal>.
+    You can change the cache timeout either in command line with 
+    <literal>--option tarball-ttl <number of seconds></literal> or
+    in the Nix configuration file with tarball-ttl
+    <literal><number of seconds to cache></literal>
```
Comment (edolstra) [inline doc/manual/expressions/builtins.xml]: Period missing.

File: doc/manual/expressions/builtins.xml [hunk 4]
```diff
@@ -280,8 +280,15 @@ with import (fetchTarball https://github.com/NixOS/nixpkgs-channels/archive/nixo
 
 stdenv.mkDerivation { … }
 </programlisting>
-
-    Note that when obtaining the hash with <varname>nix-prefetch-url
+    <para>The fetched tarball is chached for a certain amount of time
```
Comment (grahamc) [inline doc/manual/expressions/builtins.xml]: `chached` -> `cached`

File: doc/manual/expressions/builtins.xml [hunk 5]
```diff
@@ -280,8 +280,15 @@ with import (fetchTarball https://github.com/NixOS/nixpkgs-channels/archive/nixo
 
 stdenv.mkDerivation { … }
 </programlisting>
-
-    Note that when obtaining the hash with <varname>nix-prefetch-url
+    <para>The fetched tarball is chached for a certain amount of time 
+    (1 hour by default) in <literal>~/.cache/nix/tarballs/</literal>.
```
Comment (grahamc) [inline doc/manual/expressions/builtins.xml]: (a nit, you don't have to change this:) `<literal>` could be [`<filename>`](https://tdg.docbook.org/tdg/5.2/filename.html).

File: doc/manual/expressions/builtins.xml [hunk 6]
```diff
@@ -280,8 +280,15 @@ with import (fetchTarball https://github.com/NixOS/nixpkgs-channels/archive/nixo
...
+    You can change the cache timeout either on the command line with 
+    <literal>--option tarball-ttl <replaceable>number of seconds</replaceable></literal> or
```
Comment (grahamc) [inline doc/manual/expressions/builtins.xml]: (a nit, you don't have to change this:) `<literal>` could be [`<option>`](https://tdg.docbook.org/tdg/5.2/option.html), same with the following `<literal>`.

**Comment by mickours** [comment]
Done! Thanks for the review :)

File: doc/manual/expressions/builtins.xml [hunk 7]
```diff
@@ -280,8 +280,15 @@ with import (fetchTarball https://github.com/NixOS/nixpkgs-channels/archive/nixo
 
 stdenv.mkDerivation { … }
 </programlisting>
+    <para>The fetched tarball is cached for a certain amount of time
```
Comment (grahamc) [inline doc/manual/expressions/builtins.xml]: Oops, one more thing: there also needs to be a `</para>` before this opening `<para>`.

**Comment by grahamc** [comment]
There was one minor problem left, where you committed `<\para>` instead of `</para>`. I went ahead and fixed that and squashed it in to one commit. Thank you!

**Comment by grahamc** [comment]
There was one minor problem left, where you committed `<\para>` instead of `</para>`. I went ahead and fixed that and squashed it in to one commit. Thank you!

**Pull request merged**